docs/2_2_common_mechanisms.adoc

```diff
@@ -330,428 +338,335 @@ Set <<parameter,`parameters`>>, <<input,`inputs`>>, and <<start>> values, and re
+This function deserializes the byte vector `serializedState` of length `size`, constructs a copy of the FMU state and returns `FMUState`, the pointer to this copy.
+
+These functions are only supported by the FMU, if the optional capability flags `canGetAndSetFMUState` and `canSerializeFMUState` in `<fmiModelDescription><ModelExchange|CoSimulation|ScheduledExecution>` in the XML file are explicitly set to `true` (see <<fmi-for-model-exchange>>, <<fmi-for-co-simulation>>, <<fmi-for-scheduled-execution>>).
+--
+
+==== Clocks [[clocks]]
+
+FMI 3.0 specifies clearly the behavior of a clocked FMU.
+FMI 3.0 does not specify how the importer uses this functionality.
+This allows different use cases to be implemented with different semantics.
+
+===== Motivation
+
+Clock variables synchronize events between importer and across FMUs, by
+
+. communicating exactly which specific event happens and
+. exactly at which time instant, independent from continuous time specified by the arguments of `fmi3SetTime` or `fmi3DoStep`.
+
+The first point solves the issue that an FMU must know exactly which event happens during an *Event Mode* if several events are very close to one another.
+
+The second point solves the issue that computing exactly when an event happens is impossible to guarantee across importer and FMUs when using floating point computations.
+
+===== Concepts
+
+// TODO: add references to Modelica clocks paper PC06 MSL12
+// TODO: remove clockReference - clocks are variables, dependencies can be used just like for other variables
+
+This section introduces concepts related to FMI clocks.
+<<table-overview-clocks>> shows how these concepts relate.
+These concepts are described in detail after the table.
+
+// TODO: mark table as non-normative
+_[_
+
+.Overview of Clocks.
+[[table-overview-clocks]]
+[cols="10,10,15,10,40,40",options="header"]
+|====
+2+|Clock Type
+|Attribute <<causality>>
+|Attribute <<interval>>
+|Related API Calls and Arguments
+|Example
+
+.6+|time-based
+.4+|[[periodic]]periodic clock
+.4+|`input`
+|`constant`
+|[[constantClock,`constant`]]<<fmi3SetClock>> in *Event Mode*,
+<<fmi3GetInterval>> and <<fmi3GetShift>> in *Initialization Mode*
+|clocked PI-controller with a defined constant interval
+
+|`fixed`
+|[[fixedClock,`fixed`]]<<fmi3SetClock>> in *Event Mode*,
+<<fmi3SetInterval>> in *Initialization Mode*
+|clocked PI-controller with an adaptable interval
+
+|`calculated`
+|[[calculateClock,`calculated`]]<<fmi3SetClock>> in *Event Mode*,
+<<fmi3GetInterval>> and <<fmi3GetShift>> in *Initialization Mode*
+|clocked PI-controller with interval defined by fixed parameter(s) of the FMU
+
+|`tunable`
+|[[tunableClock,`tunable`]]<<fmi3SetClock>> in *Event Mode*,
+<<fmi3GetShift>> in *Initialization Mode*,
+<<fmi3GetInterval>> in *Event Mode*
+|clocked PI-controller with interval defined by tunable parameter(s) of the FMU
+
+// TODO should we rename aperiodic clocks to variable clocks to align keyword and name?
```

Comment:
I like "variable clock"

Reply:
We regrouped all aperiodic clocks in the table and renamed the `variable` clock to `changing` clock to avoid the confusion between noun and adjective `variable`.